infrastructure/libtor/src/tor.rs

```diff
@@ -95,15 +90,36 @@ impl Tor {
         instance.passphrase = TorPassword(Some(passphrase));
 
         // data dir
-        let temp = tempdir()?;
-        instance.data_dir = temp.path().to_path_buf();
-        instance.temp_dir = Some(temp);
+        instance.data_dir = data_dir.join("data");
+        match fs::exists(instance.data_dir.clone()) {
+            Ok(exists) => {
+                if !exists {
+                    if let Err(e) = fs::create_dir_all(&instance.data_dir) {
+                        return Err(ExitError::new(
+                            ExitCode::InputError,
+                            format!(
+                                "Could not create libtor data directory: {} ({})",
+                                instance.data_dir.display(),
+                                e
+                            ),
+                        ));
+                    }
+                }
+            },
+            Err(e) => {
+                return Err(ExitError::new(
+                    ExitCode::InputError,
+                    format!(
+                        "Could not verify libtor data directory: {} ({})",
+                        instance.data_dir.display(),
+                        e
+                    ),
+                ));
+            },
+        }
 
         // log destination
-        let temp = NamedTempFile::new()?.into_temp_path();
-        let file = temp.to_string_lossy().to_string();
-        instance.temp_file = Some(temp);
-        instance.log_destination = file;
+        instance.log_destination = data_dir.join("tor.log").as_path().to_string_lossy().to_string();
```

Comment:
Suggest just making log_destination a PathBuf - if someone uses non-utf8 characters in the path, this path will silently be incorrect.